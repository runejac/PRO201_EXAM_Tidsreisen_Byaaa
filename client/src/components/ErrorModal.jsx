import React from "react";
import { Modal } from "react-bootstrap";
import { CapsuleButtonYellow } from "./CapsuleButton.jsx";

export function ErrorModal({ error }) {
  return (
    <Modal
      className={"d-flex justify-content-center align-items-center"}
      show={true}
    >
      <Modal.Body>
        Woops! Det ser ut som det skjedde en feil, forsøk å klikk på "Last inn"
        eller lukk appen og start den på nytt. <br /> Teknisk melding: {error}
      </Modal.Body>
      <Modal.Footer className={"d-flex justify-content-center my-modal"}>
        <CapsuleButtonYellow
          className={"justify-content-center align-content-center"}
          buttonText={"Last inn"}
          onClick={() => window.location.reload()}
        />
      </Modal.Footer>
    </Modal>
  );
}
