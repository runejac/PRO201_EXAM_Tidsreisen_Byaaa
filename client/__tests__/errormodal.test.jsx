/**
 * @jest-environment jsdom
 */

import React from "react";
import { ErrorModal } from "../src/components/ErrorModal.jsx";
import { fireEvent, render } from "@testing-library/react";

describe("ErrorModal", () => {
  it("shows snapshot", async () => {
    const component = render(<ErrorModal />);
    expect(component).toMatchSnapshot();
  });

  it("simulates click", async () => {
    const reload = jest.fn();

    jest
      .spyOn(window, "location", "get")
      .mockImplementation(() => ({ reload }));

    const { getByText } = render(<ErrorModal />);
    fireEvent.click(getByText(/Last inn/i, { selector: "button" }));
    expect(reload).toHaveBeenCalledTimes(1);
  });

  jest.clearAllMocks();
});
