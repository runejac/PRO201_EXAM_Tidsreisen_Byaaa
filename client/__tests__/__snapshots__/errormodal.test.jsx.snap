// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal shows snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade d-flex justify-content-center align-items-center modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-body"
          >
            Woops! Det ser ut som det skjedde en feil, forsøk å klikk på "Last inn" eller lukk appen og start den på nytt. 
            <br />
             Teknisk melding: 
          </div>
          <div
            class="d-flex justify-content-center my-modal modal-footer"
          >
            <button
              id="button-capsules-yellow"
            >
              Last inn
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
